doc/_extensions/software_maturity_table.py

```diff
@@ -0,0 +1,179 @@
+        # Table information is stored in selv.env.sml_table
+        if not hasattr(self.env, "sml_table"):
+            return [
+                nodes.paragraph(text="Software maturity level table source not found")
+            ]
+
+        table_type = self.arguments[0]
+        table_info = self.env.sml_table
+
+        if table_type != "top_level" and table_type not in table_info["features"]:
+            logger.error(f"No information present for requested feature '{table_type}'")
+            return []
+
+        all_socs = table_info["all_socs"]
+        if table_type == "top_level":
+            features = table_info["top_level"]
+        else:
+            features = table_info["features"][table_type]
+
+        # Create a matrix-like table
+        table = nodes.table()
+        tgroup = nodes.tgroup(cols=len(all_socs) + 1)
+        for _ in range(len(all_socs) + 1):
+            colspec = nodes.colspec(colwidth=1)
+            tgroup.append(colspec)
+        table += tgroup
+
+        # Table head row displays the SoCs
+        thead = nodes.thead()
+        tgroup += thead
+        row = nodes.row()
+        row += nodes.entry()
+
+        for soc in all_socs:
+            soc = soc.replace("nrf", "nRF")
+            entry = nodes.entry()
+            entry += nodes.paragraph(text=soc)
+            row += entry
+        thead.append(row)
+
+        # Table body
+        rows = []
+        feature_prefix = table_type + "_"
+        for feature, maturity_list in features.items():
+            # First column displays the feature
+            row = nodes.row()
+            rows.append(row)
+            entry = nodes.entry()
+            entry += nodes.strong(text=feature.lstrip(feature_prefix))
+            row += entry
+
+            for soc in all_socs:
+                entry = nodes.entry()
+                if soc in maturity_list["Experimental"]:
+                    entry += nodes.Text("Experimental")
+                elif soc in maturity_list["Supported"]:
+                    entry += nodes.Text("Supported")
+                else:
+                    entry += nodes.Text("-")
+                row += entry
+
+        tbody = nodes.tbody()
+        tbody.extend(rows)
+        tgroup += tbody
+
+        return [table]
+
+
+def download_sml_table(app: Sphinx) -> None:
+    """Download SML table from Azure"""
+
+    # Check if local cache exists
+    if hasattr(app.env, "sml_table"):
+        local_cache = app.env.sml_table
+    else:
+        logger.info("No software maturity level cache found locally")
+        local_cache = None
+
+    # Check internet connection
+    try:
+        requests.get("https://ncsdocsa.blob.core.windows.net", timeout=3)
+    except (requests.ConnectionError, requests.Timeout):
+        logger.info("Could not retrieve software maturity level information online")
+        if local_cache:
+            logger.info("Using local software maturity level cache")
+            app.env.sml_table = local_cache
```

Comment:
this assignment seems redundant, since we have above `local_cache = app.env.sml_table`, am I missing something?

Reply:
Must be a remnant from previous logic, it is now removed